File: src/test/resources/oracle/commits/chicago-commutes-645ef20e1e3347b2d43c64a051b79830c05fe75d/android-app/src/main/java/fr/cph/chicago/util/GPSUtil.java
package fr.cph.chicago.util;

import android.location.Location;
import android.os.Looper;
import android.support.annotation.NonNull;

import com.annimon.stream.Optional;
import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.location.LocationRequest;
import com.google.android.gms.location.LocationServices;

import fr.cph.chicago.entity.Position;

public class GPSUtil {
    // The fastest refresh interval
    private static final long FASTEST_INTERVAL = 1000;
    // The minimum time between updates in milliseconds
    private static final long MIN_TIME_BW_UPDATES = 1000 * 10; // 10 sec

    private final GoogleApiClient googleApiClient;
    private final LocationRequest locationRequest;

    public GPSUtil(@NonNull final GoogleApiClient googleApiClient) {
        this.googleApiClient = googleApiClient;
        locationRequest = new LocationRequest();
        locationRequest.setInterval(MIN_TIME_BW_UPDATES);
        locationRequest.setFastestInterval(FASTEST_INTERVAL);
        locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
    }

    @NonNull
    public Optional<Position> getLocation() throws SecurityException {
        // Getting GPS status
        LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, location -> {
        }, Looper.getMainLooper());
        final Location location = LocationServices.FusedLocationApi.getLastLocation(googleApiClient);
        Position position = null;
        if (location != null) {
            position = new Position();
            position.setLatitude(location.getLatitude());
            position.setLongitude(location.getLongitude());
        }
        return Optional.ofNullable(position);
    }
}

File: src/test/resources/oracle/commits/chicago-commutes-3638be60c8bd144b968f044c0ded218e19697d69/android-app/src/main/java/fr/cph/chicago/util/GPSUtil.kt
package fr.cph.chicago.util

import android.location.Location
import android.os.Looper
import com.google.android.gms.common.api.GoogleApiClient
import com.google.android.gms.location.LocationRequest
import com.google.android.gms.location.LocationServices
import fr.cph.chicago.entity.Position

class GPSUtil(private val googleApiClient: GoogleApiClient) {
    private val locationRequest: LocationRequest = LocationRequest()

    init {
        locationRequest.interval = MIN_TIME_BW_UPDATES
        locationRequest.fastestInterval = FASTEST_INTERVAL
        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
    }

    // Getting GPS status
    val location: Position?
        @Throws(SecurityException::class)
        get() {
            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, {}, Looper.getMainLooper())
            val location: Location? = LocationServices.FusedLocationApi.getLastLocation(googleApiClient)
            return if (location == null) null else Position(location.latitude, location.longitude)
        }

    companion object {
        // The fastest refresh interval
        private val FASTEST_INTERVAL: Long = 1000
        // The minimum time between updates in milliseconds
        private val MIN_TIME_BW_UPDATES = (1000 * 10).toLong() // 10 sec
    }
}
